rmw/include/rmw/types.h

```diff
@@ -192,14 +226,40 @@ enum RMW_PUBLIC_TYPE rmw_qos_durability_policy_t
   RMW_QOS_POLICY_DURABILITY_VOLATILE
 };
 
+enum RMW_PUBLIC_TYPE rmw_qos_deadline_policy_t
+{
+  RMW_QOS_POLICY_DEADLINE_SYSTEM_DEFAULT,
+  RMW_QOS_POLICY_DEADLINE_TRANSIENT_LOCAL,
+  RMW_QOS_POLICY_DEADLINE_VOLATILE
+};
+
+enum RMW_PUBLIC_TYPE rmw_qos_liveliness_policy_t
+{
+  RMW_QOS_POLICY_LIVELINESS_SYSTEM_DEFAULT,
+  RMW_QOS_POLICY_LIVELINESS_TRANSIENT_LOCAL,
+  RMW_QOS_POLICY_LIVELINESS_VOLATILE
+};
+
+enum RMW_PUBLIC_TYPE rmw_qos_lifespan_policy_t
+{
+  RMW_QOS_POLICY_LIFESPAN_SYSTEM_DEFAULT,
+  RMW_QOS_POLICY_LIFESPAN_TRANSIENT_LOCAL,
+  RMW_QOS_POLICY_LIFESPAN_VOLATILE
+};
+
 /// ROS MiddleWare quality of service profile.
 typedef struct RMW_PUBLIC_TYPE rmw_qos_profile_t
 {
   enum rmw_qos_history_policy_t history;
   /// Size of the message queue.
   size_t depth;
+
   enum rmw_qos_reliability_policy_t reliability;
   enum rmw_qos_durability_policy_t durability;
+  enum rmw_qos_deadline_policy_t deadline;
```

Comment:
These are also going to require a number specifying the amount of time for these settings.

Reply:
@nburek  can you please elaborate?

Reply:
This is also based on an outdated version - deadline and lifespan is now an `rmw_time_t` (that's what nick meant by number)